### ISSUE TITLE
MoveItem in data provider does not take exclusions into account

### ISSUE THREAD

**Issue by mzizka** [kamsar/Unicorn#161]
I ran into this problem with the sample Habitat build. Consider a predicate with this inclusion:

```xml
<include name="Foundation.Serialization.Renderings.Project" database="master" path="/sitecore/layout/renderings/Project">
  <exclude children="true" />
</include>
```

If I move an item into `/sitecore/layout/renderings/Project`, this item will be serialized to the file system by Unicorn. Additionally, if I reorder the items within that same folder by moving one item before or after another, that item will also be serialized.

My understanding is that when an item is moved, the data provider should check if the predicate includes the **item's new path** (`/sitecore/layout/renderings/Project/Moved-Item`), not its destination (which is its new parent path, i.e. `/sitecore/layout/renderings/Project`). I was able to fix this with the following change, but I am not sufficiently confident in my Unicorn knowledge to make it a pull request:

```diff
diff --git a/src/Unicorn/Data/DataProvider/UnicornDataProvider.cs b/src/Unicorn/Data/DataProvider/UnicornDataProvider.cs
index f6f4155..25fa1cc 100644
--- a/src/Unicorn/Data/DataProvider/UnicornDataProvider.cs
+++ b/src/Unicorn/Data/DataProvider/UnicornDataProvider.cs
@@ -197,7 +197,10 @@ namespace Unicorn.Data.DataProvider
 
 			if (destinationItem == null) return; // can occur with TpSync on, when this isn't the configuration we're moving for the data store will return null
 
-			if (!_predicate.Includes(destinationItem).IsIncluded) // if the destination we are moving to is NOT included for serialization, we delete the existing item
+            // rebase the path to the new destination path (this handles children too)
+            var rebasedSourceItem = new PathRebasingProxyItem(sourceItem, destinationItem.Path, destinationItem.Id);
+
+            if (!_predicate.Includes(rebasedSourceItem).IsIncluded) // if the destination we are moving to is NOT included for serialization, we delete the existing item
 			{
 				var existingItem = _targetDataStore.GetByPathAndId(sourceItem.Path, sourceItem.Id, sourceItem.DatabaseName);
 
@@ -210,9 +213,6 @@ namespace Unicorn.Data.DataProvider
 				return;
 			}
 
-			// rebase the path to the new destination path (this handles children too)
-			var rebasedSourceItem = new PathRebasingProxyItem(sourceItem, destinationItem.Path, destinationItem.Id);
-
 			// this allows us to filter out any excluded children by predicate when the data store moves children
 			var predicatedItem = new PredicateFilteredItemData(rebasedSourceItem, _predicate);
 
```

**Comment by kamsar** [comment]
Good catch and well documented.

**Comment by kamsar** [comment]
I have tested your proposed solution and think it works just fine. It'll be in the next release. Thanks for the diagnostics and fixed code :)

**Comment by kamsar** [comment]
Released in 3.3.1
